seatunnel-connectors/seatunnel-connector-flink-iotdb/src/main/java/org/apache/seatunnel/flink/source/IotDbInputFormat.java

```diff
@@ -0,0 +1,190 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.flink.source;
+
+import org.apache.flink.api.common.io.DefaultInputSplitAssigner;
+import org.apache.flink.api.common.io.RichInputFormat;
+import org.apache.flink.api.common.io.statistics.BaseStatistics;
+import org.apache.flink.api.java.typeutils.ResultTypeQueryable;
+import org.apache.flink.api.java.typeutils.RowTypeInfo;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.io.GenericInputSplit;
+import org.apache.flink.core.io.InputSplit;
+import org.apache.flink.core.io.InputSplitAssigner;
+import org.apache.flink.types.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.List;
+
+public class IotDbInputFormat extends RichInputFormat<Row, InputSplit> implements ResultTypeQueryable<Row> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(IotDbInputFormat.class);
+
+    private static final String IOTDB_DRIVER = "org.apache.iotdb.jdbc.IoTDBDriver";
+
+    private String url;
+    private String query;
+    private List<String> fields;
+    private RowTypeInfo rowTypeInfo;
+
+    private transient Connection connection;
+    private transient Statement statement;
+
+    private long offset;
+    private boolean hasNext;
+
+    public IotDbInputFormat() {
+    }
+
+    @Override
+    public RowTypeInfo getProducedType() {
+        return rowTypeInfo;
+    }
+
+    @Override
+    public void configure(Configuration parameters) {
+    }
+
+    @Override
+    public void openInputFormat() {
+        try {
+            Class.forName(IOTDB_DRIVER);
+            connection = DriverManager.getConnection(url, "root", "root");
+            statement = connection.createStatement();
+            offset = 0;
+            hasNext = true;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("openInputFormat() failed." + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void closeInputFormat() {
+        try {
+            if (connection != null) {
+                connection.close();
+                connection = null;
+            }
+        } catch (Exception e) {
+            LOG.error("IotDbInputFormat couldn't be closed - " + e.getMessage());
```

Comment:
consider using this so that the exception stack can be exposed

```suggestion
            LOG.error("IotDbInputFormat couldn't be closed", e);
```

Reply:
Agree, many other plgins can be improved too, I originally thought it was coding style :sweat_smile:

Reply:
Agree, many other plugins can be improved too, I originally thought it was coding style :sweat_smile:

Reply:
@kezhenxu94 Would you like raise a issue to describe it, and I will fix them in another PR

Reply:
https://github.com/apache/incubator-seatunnel/issues/1291

Reply:
Cool, thanks a lot

Reply:
You can do this in another PR, I'll mark this as resolved

Reply:
@kezhenxu94 Sure